rainmaker_orchestrator/core/orchestrator.py

```
@@ -0,0 +1,405 @@
+                mode='w',
+                suffix='.py',
+                dir=self.workspace_path,
+                delete=False
+            ) as f:
+                f.write(code)
+                temp_file = Path(f.name)
+
+            try:
+                # Execute using subprocess
+                env = environment.copy() if environment else {}
+                result = subprocess.run(
```

Ruff (0.14.10)

`subprocess` call: check for execution of untrusted input

(S603)